### PR TITLE
Add Alerting V2

### PR DESCRIPTION
**Pull request by aiyengar2**

### CODE REVIEW

**Comment by aiyengar2** [comment]
Slightly modified the default template and added a note to the `CHANGELOG.md` in the latest push

File: packages/rancher-monitoring/rancher-monitoring.patch [hunk 1]
```diff
@@ -1544,7 +1544,100 @@ diff -x '*.tgz' -x '*.lock' -uNr packages/rancher-monitoring/charts-original/val
...
++        {{- if .Labels.severity }}
++        *Severity:* `{{ .Labels.severity }}`
++        {{- end }}
++        {{- if .Annotations.description }}
++        *Description:* {{ .Annotations.description }}
++        {{- end }}
++        {{- if .Annotations.runbook }}
++        *Runbook:* <{{ .Annotations.runbook }}|:spiral_note_pad:>
++        {{- end }}
++        *Details:*
```
Comment (cbron) [inline packages/rancher-monitoring/rancher-monitoring.patch:1633]: No if check here ? To see if Labels.sortedPairs is empty.

File: packages/rancher-monitoring/rancher-monitoring.patch [hunk 2]
```diff
@@ -1544,7 +1544,100 @@ diff -x '*.tgz' -x '*.lock' -uNr packages/rancher-monitoring/charts-original/val
...
+-  #           {{ end }}
+-  #       {{ end }}
+-  #       {{ end }}
++  templateFiles:
++    rancher_defaults.tmpl: |-
++        {{- define "slack.rancher.text" -}}
++        {{ template "rancher.text_multiple" . }}
++        {{- end -}}
++
++        {{ define "cluster" }}{{ .ExternalURL | reReplaceAll ".*alertmanager\\.(.*)" "$1" }}{{ end }}
```
Comment (cbron) [inline packages/rancher-monitoring/rancher-monitoring.patch:1608]: nit: other examples are on separate lines not on one line

File: packages/rancher-monitoring/rancher-monitoring.patch (hunk 1 above)
Comment (aiyengar2) [reply packages/rancher-monitoring/rancher-monitoring.patch:1636]: Good catch! Updated in latest commit

File: packages/rancher-monitoring/rancher-monitoring.patch (hunk 2 above)
Comment (aiyengar2) [reply packages/rancher-monitoring/rancher-monitoring.patch]: Fixed in latest commit

**Pull request merged**